pkg/proof/README.md

```diff
@@ -0,0 +1,198 @@
+
+For instance, if we want to prove the first two shares, we will need two merkle proofs:
+
+<img src="img/share_to_share_commitment_inclusion.png" alt="shares to share commitment inclusion proof" width="500"/>
+
+- The first merkle proof is a namespace merkle inclusion proof from the first two shares to the *subtree root* `SR1`.
+- The second merkle tree is a binary merkle inclusion proof from `SR1`,
+  which is the subtree root committing to the shares, to the *share commitment*.
+
+Note: the nodes colored in brown are the inner nodes used to construct the inclusion proof.
+
+## Square layout
+
+Now that the *share commitment* is generated.
+The transaction gets broadcasted to the Celestia network to be picked up by validators.
+Once it's included in a block, the transaction, without the blob,
+is placed in the [transaction namespace](https://celestiaorg.github.io/celestia-app/specs/namespace.html#reserved-namespaces)
+and the blob is placed in the [namespace](https://github.com/celestiaorg/celestia-app/blob/72be251f044bcece659603248bc27711b2c039a0/proto/celestia/blob/v1/tx.proto#L22)
+specified in the PFB transaction.
+
+If we use the same transaction from above,
+where the blob consists of six shares,
+an example Celestia square containing the blob and the PFB transaction can look like this:
+
+<img src="img/extended_square.png" alt="celestia extended square containing the blob shares" width="500"/>
+
+The share range `[5, 10]`, colored in green, is the blob data.
+
+### Row roots
+
+To compute the Celestia block [data root](https://celestiaorg.github.io/celestia-app/specs/data_structures.html),
+we first start by computing the row roots and column roots.
+
+For instance, if we want to compute the first row root:
+
+<img src="img/row_root_1.png" alt="row root one generation" width="500"/>
+
+We use the first row of shares to generate a namespace merkle root.
+The first four shares, i.e., the shares `[1, 4]` are part of the square data,
+and the last four are [parity shares](https://celestiaorg.github.io/celestia-app/specs/data_structures.html?highlight=extended#2d-reed-solomon-encoding-scheme). 
+
+Now if we take the second row, which contains four shares of the above blob `[5,8]`:
+
+<img src="img/row_root_2.png" alt="row root two generation" width="500"/>
+
+The inner node in blue is the same as the *subtree root* `SR1`
+generated in the [PFB transaction](#pfb-transaction) section. 
+
+Similarly, if we generate the third row, which contains the remaining two shares of the blob, and some other data:
+
+<img src="img/row_root_3.png" alt="row root three generation" width="500"/>
+
+We will see that the *subtree root* `SR2` is an inner node of the tree used to compute the third row root.
+
+This property holds for all the subtree roots
+computed for the blob data
+and is derived
+from applying [non-interactive defaults](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html) rules
```

Comment:
IMO we should use blob share commitment rules instead of non-interactive defaults.

```suggestion
from applying [Blob Share Commitment Rules](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
```